### PR TITLE
Fix for nonexistant karma record, with pre-failing test

### LINKED ISSUE #1630: Commits associated with DSCP Partner Portal not showing up in karma
For @freeranger and potentially others.  

I also note that the commit count for the project is still zero.  

Relevant code from GithubCommitsJob:

```
def run
    Project.with_github_url.each do |project|
      begin
        update_total_commit_count_for(project)
        update_user_commit_counts_for(project)
      rescue StandardError
        Rails.logger.warn "#{project.github_url} may have caused the issue. Commit1 update terminated for this project!"
      end
    end
  end
```

If update_total_commit_count_for(project) errors out in Octokit call due to rate limiting then subsequent user commit update will never be run either.

### PR DESCRIPTION
**Pull request by mattlindsey**
fixes #1630

### CODE REVIEW

File: app/services/karma_calculator.rb [hunk 1]
```diff
@@ -7,9 +7,13 @@ def initialize(user)
   end
 
   def perform
-    user.karma.total = 0
-    return if user.created_at.blank?
+    if user.karma
+      user.karma.total = 0
+    else
+      user.karma = Karma.find_or_create_by(user_id: user.id, total: 0)
+    end
```
Comment (tansaku) [inline app/services/karma_calculator.rb]: could simplify to 

```rb
Karma.find_or_create_by(user_id: user.id, total: 0)
user.karma.total = 0
```
but well ...

Comment (tansaku) [inline app/services/karma_calculator.rb]: @mattlindsey let me know if you want to make a change here, or leave to @mtc2013 to take a refactoring pass?

Comment (mattlindsey) [inline app/services/karma_calculator.rb]: @tansaku I would like to leave it as-is for now, since I tested it.  @mtc2013 can refactor.

**Comment by mtc2013** [comment]
@tansaku @mattlindsey someone should kick this build off again.

**Comment by mattlindsey** [comment]
@tansaku I no longer have permission in Semaphore to rerun a build. Could you kick this off and/or give me permission.

**Comment by tansaku** [comment]
@mattlindsey apologies - have kicked off build - also tried to give you permissions again

File: app/services/karma_calculator.rb (hunk 1 above)
Comment (tansaku) [inline app/services/karma_calculator.rb]: ideally the presence of tests should give you confidence to refactor :-) but fine to leave for @mtc2013 ...

**Pull request merged**